crates/ruff_db/src/files/file_root.rs

```diff
@@ -0,0 +1,119 @@
+use std::fmt::Formatter;
+
+use path_slash::PathExt;
+
+use crate::file_revision::FileRevision;
+use crate::system::{SystemPath, SystemPathBuf};
+use crate::Db;
+
+/// A root path for files tracked by the database.
+///
+/// We currently create roots for:
+/// * static module resolution paths
```

Comment:
What's the reason for not doing so for editable-install search paths?

Reply:
Hmm. Not sure. I think I don't understand them well enough. 

What I understand is that editable installs are for example used to make workspace packages available to other packages. Or to link in any external package. The first is somewhat problematic because the path overlaps with the package path but we definitely want durability low for first party package files. 

For both cases I think what we want is durability low which will be the default for all files for which we can't find a file root.

Reply:
I'm landing this PR as is. I can create roots for editable installs when needed as part of a separate PR

Reply:
Yes, fine by me!